CON-40 allow replica node to have separate wal segments location

patagona-anas opened the pull request.

patagona-anas committed:
CON-40 set wal directory to /wal location
It enables replica nodes to have different wal segments location

patagona-anas committed:
CON-40 update build guideline for generating spilo

mzahn86 commented on postgres-appliance/docker-compose.yaml:
```diff
@@ -0,0 +1,15 @@
+        # set to true if you want to install perl and plperl packages into image
+        WITH_PERL: 'false'
+        PGVERSION: '13'
+        PGOLDVERSIONS: '9.5 9.6 10 11 12'
```
can we set this to empty? then the building will be faster since it only needs to be built against the most recent postgres version

patagona-anas replied:
I will check later but first I can check with these values

mzahn86 commented on postgres-appliance/docker-compose.yaml:
```diff
@@ -0,0 +1,15 @@
+version: "3.5"
+services:
+  spilo:
```
how do we specifiy timescaledb?

patagona-anas replied:
I have added the one

mzahn86 commented on postgres-appliance/docker-compose.yaml:
```diff
@@ -0,0 +1,15 @@
+        # set to true to build the smallest possible image which will work only on Kubernetes
+        DEMO: 'false'
+        # set to false to build timescaledb community version (Timescale License)
+        TIMESCALEDB_APACHE_ONLY: 'true'
```
what have we used for our current cluster?

patagona-anas replied:
It's false. Adjusted. I made mistake while copying this env

mzahn86 commented on postgres-appliance/docker-compose.yaml:
```diff
@@ -0,0 +1,15 @@
+      args:
+        # set to true if you want to install perl and plperl packages into image
+        WITH_PERL: 'false'
+        PGVERSION: '13'
```
do we need to provide the exact version of our cluster? rephrase: is a different postgres minor version compatible with our cluster?

patagona-anas replied:
Yes they are backward compatible [1] but I would still like to use 13.3 version (same as we use in production). 

I am not sure how can I use it with current docker file

[1] https://www.postgresql.org/docs/13/upgrading.html

patagona-anas replied:
UPDATE - I tried to use 13.3 and building image failed due to missing packages. I think I need to use 13. 

After successful building of image, I will use new postgres 13.x version in our patagona-api and see it everything works as expected.

```
Building dependency tree...
Reading state information...
E: Unable to locate package postgresql-contrib-13.3
E: Couldn't find any package by glob 'postgresql-contrib-13.3'
E: Couldn't find any package by regex 'postgresql-contrib-13.3'
E: Unable to locate package postgresql-plpython3-13.3
E: Couldn't find any package by glob 'postgresql-plpython3-13.3'
E: Couldn't find any package by regex 'postgresql-plpython3-13.3'
E: Unable to locate package postgresql-server-dev-13.3
E: Couldn't find any package by glob 'postgresql-server-dev-13.3'
E: Couldn't find any package by regex 'postgresql-server-dev-13.3'
E: Unable to locate package postgresql-13.3-cron
E: Couldn't find any package by glob 'postgresql-13.3-cron'
E: Couldn't find any package by regex 'postgresql-13.3-cron'
E: Unable to locate package postgresql-13.3-pgq3
E: Couldn't find any package by glob 'postgresql-13.3-pgq3'
E: Couldn't find any package by regex 'postgresql-13.3-pgq3'
E: Unable to locate package postgresql-13.3-pg-stat-kcache
E: Couldn't find any package by glob 'postgresql-13.3-pg-stat-kcache'
```

patagona-anas committed:
CON-40 wip

https://github.com/zalando/spilo/commit/c56d01adf5f7845c0abccc792fdc99ef8ceb8d0d

patagona-anas commented:
I think docker building images is already broken and it has nothing to do with my changes. If you see here [1], I have just modified README and building docker image still fail [2].

[1] https://github.com/Patagona/spilo/pull/6/files
[2] https://github.com/Patagona/spilo/actions/runs/3091330175

patagona-anas commented:
Since our base image is using ubuntu 18.04 so maybe its due to this deprecation notice [1]

[1] https://github.com/actions/runner-images/issues/6002
